stdlib/src/collections/index_normalization.mojo

```diff
@@ -0,0 +1,81 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+"""The utilities provided in this module help normalize the access 
+to data elements in arrays."""
+
+
+fn _get_out_of_bounds_error_message[
+    container_name: String
+](i: Int, container_length: Int) -> String:
+    if container_length == 0:
+        return (
+            "The "
+            + container_name
+            + " has a length of 0. "
+            + "Thus it's not possible to access its values with an index "
+            + "but the index value "
+            + str(i)
+            + " was used. "
+            + "Aborting now to avoid an out-of-bounds access."
+        )
+    else:
+        return (
+            "The "
+            + container_name
+            + " has a length of "
+            + str(container_length)
+            + ". "
+            + "Thus the index provided should be between "
+            + str(-container_length)
+            + " (inclusive) and "
+            + str(container_length)
+            + " (exclusive) but the index value "
+            + str(i)
+            + " was used. "
+            + "Aborting now to avoid an out-of-bounds access."
+        )
+
+
+@always_inline
+fn normalize_index[
```

Comment:
This is a nice utility, but I don't think it should be a public method. In fact, I'm not sure this should be a public module at all. WDYT about renaming to `_index_normalization`?

Reply:
It's not very clear what is public/private in the stdlib. I added an underscore to the module name as asked.